File: backend/models/user.model.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

//connection
require('../database/mongoose')

// creating the collection
const UserSchema = mongoose.Schema({
    name : { type: String, required: [true, "Please fill this field"] },
    username : { type: String, required: [true, "Please fill this field"] },
    email : { type: String, required: [true, "Please fill this field"], unique: [true, "Already taken"] },
    password : { type: String, required: [true, "Please fill this field"] },
    phone : { type: String, required: [true, "Please fill this field"] },
    location : { type: String, required: [true, "Please fill this field"] },
    languages : [ { type: String } ],
    hobbies : [ { type: String } ],
    interests : [ { type: String } ],
    experiance : [{ type: Schema.Types.ObjectId, ref: 'Experiance' }],
    age : { type: Number },
    photo : { type: String, },
    status : { type: Boolean, default: true },
    type : { type: String, default: "basic" },
    favorites : [{ type: Schema.Types.ObjectId, ref: this }]
});

// to create data objects based on the schema we need model
module.exports = User = mongoose.model('User',UserSchema);

File: frontend/lib/models/User.model.dart

class User {
  String id;
  String name;
  String username;
  String password;
  String email;
  String phone;
  String location;
  String type;
  String photo;
  bool status;
  List<String> languages;
  List<String> hobbies;
  List<String> interests;
  List<dynamic> experiences;

  User({
    required this.id,
    required this.name,
    required this.username,
    required this.email,
    required this.phone,
    required this.password,
    required this.location,
    required this.languages,
    required this.interests,
    required this.experiences,
    required this.hobbies,
    required this.status,
    required this.type,
    required this.photo
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
        id: json['_id'],
        name: json['name'],
        username: json['username'],
        email: json['email'],
        phone: json["phone"],
        password: json["password"],
        location: json["location"],
        status: json["status"],
        type: json["type"],
        photo: json["photo"],
        languages:  List<String>.from(json["languages"]),
        experiences: json["experience"],
        hobbies:  List<String>.from(json["hobbies"]),
        interests:  List<String>.from(json["interests"]),
    );
  }
}
